src/source/Rtp/Codecs/RtpH265Payloader.c

```diff
@@ -0,0 +1,269 @@
+        payloadArray.payloadSubLenSize = 0;
+    }
+
+    if (pPayloadSubLenSize != NULL && pPayloadLength != NULL) {
+        *pPayloadLength = payloadArray.payloadLength;
+        *pPayloadSubLenSize = payloadArray.payloadSubLenSize;
+    }
+
+    LEAVES();
+    return retStatus;
+}
+
+STATUS getNextNaluLengthH265(PBYTE nalus, UINT32 nalusLength, PUINT32 pStart, PUINT32 pNaluLength)
+{
+    ENTERS();
+
+    STATUS retStatus = STATUS_SUCCESS;
+    UINT32 zeroCount = 0;
+    UINT32 offset;
+    BOOL naluFound = FALSE;
+    PBYTE pCurrent = NULL;
+
+    CHK(nalus != NULL && pStart != NULL && pNaluLength != NULL, STATUS_NULL_ARG);
+
+    // Annex-B Nalu will have 0x000000001 or 0x000001 start code, at most 4 bytes
+    for (offset = 0; offset < 4 && offset < nalusLength && nalus[offset] == 0; offset++)
+        ;
+
+    CHK(offset < nalusLength && offset < 4 && offset >= 2 && nalus[offset] == 1, STATUS_RTP_INVALID_NALU);
+    *pStart = ++offset;
+    pCurrent = nalus + offset;
+
+    /* Not doing validation on number of consecutive zeros being less than 4 because some device can produce
+     * data with trailing zeros. */
+    while (offset < nalusLength) {
+        if (*pCurrent == 0) {
+            /* Maybe next byte is 1 */
+            offset++;
+            pCurrent++;
+
+        } else if (*pCurrent == 1) {
+            if (*(pCurrent - 1) == 0 && *(pCurrent - 2) == 0) {
+                zeroCount = *(pCurrent - 3) == 0 ? 3 : 2;
+                naluFound = TRUE;
+                break;
+            }
+
+            /* The jump is always 3 because of the 1 previously matched.
+             * All the 0's must be after this '1' matched at offset */
+            offset += 3;
+            pCurrent += 3;
+        } else {
+            /* Can jump 3 bytes forward */
+            offset += 3;
+            pCurrent += 3;
+        }
+    }
+    *pNaluLength = MIN(offset, nalusLength) - *pStart - (naluFound ? zeroCount : 0);
+
+CleanUp:
+
+    // As we might hit error often in a "bad" frame scenario, we can't use CHK_LOG_ERR as it will be too frequent
+    if (STATUS_FAILED(retStatus)) {
+        DLOGD("Warning: Failed to get the next NALu in H265 payload with 0x%08x", retStatus);
+    }
+
+    LEAVES();
+    return retStatus;
+}
+
+STATUS createPayloadFromNaluH265(UINT32 mtu, PBYTE nalu, UINT32 naluLength, PPayloadArray pPayloadArray, PUINT32 filledLength,
+                                 PUINT32 filledSubLenSize)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    PBYTE pPayload = NULL;
+    UINT8 naluType = 0;
+    UINT32 maxPayloadSize = 0;
+    UINT32 curPayloadSize = 0;
+    UINT32 remainingNaluLength = naluLength;
+    UINT32 payloadLength = 0;
+    UINT32 payloadSubLenSize = 0;
+    PBYTE pCurPtrInNalu = nalu;
+    BOOL sizeCalculationOnly = (pPayloadArray == NULL);
+
+    CHK(nalu != NULL && filledLength != NULL && filledSubLenSize != NULL, STATUS_NULL_ARG);
+    sizeCalculationOnly = (pPayloadArray == NULL);
```

Comment:
duplicate line, you have this already 3 lines above.

Reply:
Removed from both, h264 and h265